edgedb/options.py

```diff
@@ -235,24 +312,70 @@ def with_state(self, state: State):
         result._options = self._options.with_state(state)
         return result
 
-    def with_module_aliases(self, module=None, **aliases):
+    def with_default_module(self, module: typing.Optional[str] = None):
+        result = self._shallow_clone()
+        result._options = self._options.with_state(
+            self._options.state.with_default_module(module)
+        )
+        return result
+
+    def with_module_aliases(
+        self,
+        aliases_dict: typing.Optional[typing.Mapping[str, str]] = None,
+        **aliases,
+    ):
+        result = self._shallow_clone()
+        result._options = self._options.with_state(
+            self._options.state.with_module_aliases(aliases_dict, **aliases)
+        )
+        return result
+
+    def with_config(
+        self,
+        config_dict: typing.Optional[typing.Mapping[str, typing.Any]] = None,
+        **config,
+    ):
+        result = self._shallow_clone()
+        result._options = self._options.with_state(
+            self._options.state.with_config(config_dict, **config)
+        )
+        return result
+
+    def with_globals(
+        self,
+        globals_dict: typing.Optional[typing.Mapping[str, typing.Any]] = None,
```

Comment:
Can we give `globals_dict` a dodgier name? Like `__globals_dict`? Obviously the best thing here would be to make it positional only, but that doesn't work on 3.7.

The other way to handle this "correctly" would be to take `*args` and `**kwargs` and manually extract the argument from `*args` (and probably produce an error if there is more than one arg). Probably not worth bothering, though.